crates/bevy_ecs/src/system/system.rs

```diff
@@ -61,12 +61,46 @@ pub trait System: Send + Sync + 'static {
     ///        [`System::archetype_component_access()`].
     unsafe fn run_unsafe(&mut self, input: Self::In, world: &World) -> Self::Out;
     /// Runs the system with the given input in the world.
+    ///
+    /// Use [`run_direct`] instead if you are manually running a system outside of a schedule
     fn run(&mut self, input: Self::In, world: &mut World) -> Self::Out {
         // SAFE: world and resources are exclusively borrowed
         unsafe { self.run_unsafe(input, world) }
     }
+    /// Runs the system directly on the world, initializing the world correctly;
+    /// immediately applying buffers (such as `Commands`) modified by its system parameters
+    ///
+    /// Use () as the `input` parameter for systems which do not take any chained input.
+    ///
+    /// Only one system will run at a time when executed in this way;
+    /// use a [`Schedule`] (or a custom abstraction created with [`run_unsafe`])
+    /// when system parallelism is desired.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bevy_ecs::prelude::*;
+    ///
+    /// struct Counter(u8);
+    /// let mut world = World::new();
+    ///
+    /// fn count_up(mut counter: ResMut<Counter>){
+    ///     counter.0 += 1;
+    /// }
+    ///
+    /// world.insert_resource::<Counter>(Counter(0));
+    /// count_up.system().run_direct((), &mut world);
```

Comment:
I couldn't get this to work without `.system()`:

```
  method not found in `for<'r> fn(bevy_ecs::change_detection::ResMut<'r, Counter>) {count_up}`
```
I would like to tackle this to allow us to deprecate `.system()` but I'm not sure where to start looking; especially when we want to persistently store these system-ized functions in a resource somewhere.

Reply:
I think you'll need to add a new extension trait for systems, something like `DirectlyRunnableSystem` and then impl it for `IntoSystem`

For example, for the `.config` method:
* new trait:
https://github.com/bevyengine/bevy/blob/b52edc107de5615e92662bb860bfe7831a15a6f1/crates/bevy_ecs/src/system/function_system.rs#L268-L276
* implementation for `IntoSystem`:
https://github.com/bevyengine/bevy/blob/b52edc107de5615e92662bb860bfe7831a15a6f1/crates/bevy_ecs/src/system/function_system.rs#L278-L292

Reply:
Done! I enabled system-yeet for run, initialize, and apply_buffers as well.